ui/views.py

```diff
@@ -240,7 +242,12 @@ def extra_context(self):
 @login_required
 def export(request, resource_id):
     """Dump LearningResource as XML"""
-    return HttpResponse(
-        get_resource(resource_id, request.user.id).content_xml,
-        content_type='text/xml'
-    )
+    try:
+        return HttpResponse(
+            get_resource(resource_id, request.user.id).content_xml,
+            content_type='text/xml'
+        )
+    except NotFound:
+        raise Http404()
+    except PermissionDenied:
+        raise DjangoDenied
```

Comment:
Evidently it passes tests either way. I added the `()` for consistency. Stand by for another push when tests pass.